### ISSUE TITLE
[BUG] Received "document is not defined" when using Cheerio, Playwright, and Puppeteer Web Scrapers

### ISSUE THREAD

**Issue by FuadAbdullah** [FlowiseAI/Flowise#3670]
**Describe the bug**
When I am testing Cheerio, Playwright, and Puppeteer to scrape AWS documentation into a document, I receive an error saying that a document is not defined and then was returned the exact same link which I have provided initially to the node. I have tested with two links, with one meant for web scraping tests but to no avail.

**To Reproduce**

1. Go to 'Chatflows' 
2. Add a new Chatflow
3. Add any of the three web crawler node (Cheerio, Playwright, Puppeteer)
4. Paste in any of the two URLs I have provided in the Screenshot section below
5. Click on 'Manage Links'
6. Click on 'Fetch Links'
7. Observe that the 'Scraped Links' returns the exact same link provided early on
8. Observe FlowiseAI Docker console for the error log (with DEBUG environment variable set to `true`)

**Expected behavior**
I expect multiple other links beside the initial link that I provide coming from the page I intended to scrape (AWS Documentation and Web Scraper test site)

**Screenshots**
Link: https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_PerfInsights.html
![image](https://github.com/user-attachments/assets/840d0ecc-f34d-44a1-990b-d59fcc8102cc)
![image](https://github.com/user-attachments/assets/c50ec02e-a750-444c-a61f-924d209f398d)

Link: https://webscraper.io/test-sites/e-commerce/static
![image](https://github.com/user-attachments/assets/e4375186-ce5c-4a36-b166-55e7a5ee944f)
![image](https://github.com/user-attachments/assets/51acfcbd-0028-4c5e-a3a1-025149e24c2e)

Error logs through console:
![image](https://github.com/user-attachments/assets/b8f4fa0a-a734-4e19-898b-c06e7f2e0b29)

**Flow**
Exported flow for reference (JSON):
[Bug Report Flow Chatflow.json](https://github.com/user-attachments/files/18079498/Bug.Report.Flow.Chatflow.json)

**Setup**

-   Installation: Docker
-   Flowise Version: 	flowise@2.2.1
-   OS: Linux
-   Browser: Chrome

**Additional context**
I have peeked at the codebase and explored the following files to identify where the issue could be stemming from:
https://github.com/FlowiseAI/Flowise/blob/main/packages/server/src/services/fetch-links/index.ts
https://github.com/FlowiseAI/Flowise/blob/main/packages/components/src/utils.ts#L349

**Comment by HenryHengZJ** [comment]
Works for me though when using document store, can you try it?

![image](https://github.com/user-attachments/assets/88d5ad64-4e70-42ec-807a-b53ff3d54cdd)

**Comment by FuadAbdullah** [comment]
Hi Henry, that is really weird. Let me try again now and see if it makes any difference.

**Comment by FuadAbdullah** [comment]
Hi Henry, I can confirm it is now working for both in Document Store and in a Chatflow:

![image](https://github.com/user-attachments/assets/a50f4a56-2304-47b0-aa31-1539dd9a85bb)

![image](https://github.com/user-attachments/assets/3773b671-5b10-4238-bc9c-63c0cb193fff)

Thanks for helping out, appreciate it!
